src/contrib/cluster/Akka.Cluster.Sharding/ShardedDaemonProcess.cs

```diff
@@ -150,29 +203,71 @@ public void Init(string name, int numberOfInstances, Func<int, Props> propsFacto
                 TimeSpan.Zero, // passivation disabled
                 StateStoreMode.DData,
                 shardingBaseSettings.TuningParameters,
-                shardingBaseSettings.CoordinatorSingletonSettings, 
+                shardingBaseSettings.CoordinatorSingletonSettings,
                 shardingBaseSettings.LeaseSettings);
 
-            if (string.IsNullOrEmpty(shardingSettings.Role) || Cluster.Get(_system).SelfRoles.Contains(shardingSettings.Role))
+            if (string.IsNullOrEmpty(shardingSettings.Role) ||
+                Cluster.Get(_system).SelfRoles.Contains(shardingSettings.Role))
             {
                 var sharding = ClusterSharding.Get(_system);
                 var shardingRef = sharding.Start(
-                    typeName: $"sharded-daemon-process-{name}",
+                    typeName: FormatWorkerProcessName(name),
```

Comment:
Use a shared function to name `ShardedDaemonProcess`es and proxies - uses the exact same lexical structure as before so it won't break any existing deployments.